rocketpool/node/collectors/node-collector.go

```diff
@@ -292,6 +294,19 @@ func (collector *NodeCollector) Collect(channel chan<- prometheus.Metric) {
 			return err
 		}
 
+		// Get the totalNodeWeight for the last completed interval
+		previousRewardIndex := state.NetworkDetails.RewardIndex - 1
+		previousInterval, err := rprewards.GetIntervalInfo(collector.rp, collector.cfg, collector.nodeAddress, previousRewardIndex, nil)
+		if err != nil {
+			return err
+		}
+
+		if !previousInterval.TreeFileExists {
+			return fmt.Errorf("Error retrieving previous interval's total node weight: rewards file %s doesn't exist for interval %d", previousInterval.TreeFilePath, previousRewardIndex)
+		}
+		// Convert to a float, accuracy loss is meaningless compared to the heuristic's natural inaccuracy.
+		previousIntervalTotalNodeWeight = &previousInterval.TotalNodeWeight.Int
```

Comment:
Protect against div-by-0 if TotalNodeWeight is 0

Reply:
Done by verifying nodeWeight is > 0 before calculating estimatedRewards. Since we divide the nodeWeight by the sum of the nodeWeight + previousIntervalTotalNodeWeight, this should ensure that we never divide by 0.

Reply:
ok but if my node weight is 100 and and previousInterval is 0 the predictor also breaks and says you'll get 100% of the rewards

Reply:
I'll just check if it's the first interval and if it is return 0 for the estimate